### PR TITLE
Prefix most root variables with `$kendo-`

### PR DESCRIPTION
**Pull request by joneff**
Note: The commits are separate, so I can edit them easier.

There should be one image difference -- I've merged the styles of `k-placeholder-line` with `k-skeleton-text`, because we already have it.

After the PR critique is resolved, I will squash the commits into two -- one for the breaking change, listing all variable changes; one for the docs.

### CODE REVIEW

File: packages/default/scss/common/_base.scss [hunk 1]
```diff
@@ -68,45 +68,11 @@
...
-
-        > .k-header {
-            @include border-top-radius( $kendo-border-radius-md );
-            @include fill(
-                $header-text,
-                $header-bg,
-                $header-border,
-                $header-gradient
-            );
-            margin: #{-$panel-padding-y} #{-$panel-padding-x} $panel-padding-y;
```
Comment (Juveniel) [inline packages/default/scss/common/_base.scss:98]: where did those style go ? they are still being used across components

File: packages/default/scss/wizard/_variables.scss [hunk 2]
```diff
@@ -1,14 +1,13 @@
...
+$wizard-content-padding-y: $kendo-padding-md-y * 8 !default;
 $wizard-buttons-margin-x: 10px !default;
 
 $wizard-border-width: 0px !default;
 $wizard-font-size: $font-size !default;
 $wizard-line-height: $line-height !default;
 $wizard-font-family: $font-family !default;
 
-$wizard-step-border-focused: #656565 !default;
-$wizard-focused-shadow: inset 0 0 0 2px rgba(0, 0, 0, .13) !default;
```
Comment (Juveniel) [inline packages/default/scss/wizard/_variables.scss:14]: is the box-shadow var removed by accident here ?

File: packages/fluent/scss/core/_variables.scss [hunk 3]
```diff
@@ -95,10 +91,12 @@ $kendo-border-radius-lg: ( $kendo-border-radius * 2 ) !default;
...
 $kendo-padding-x: map.get( $kendo-spacing, 2 ) !default;
 $kendo-padding-y: map.get( $kendo-spacing, 1 ) !default;
-$kendo-padding-x-sm: math.div( $kendo-padding-x, 2) !default;
-$kendo-padding-y-sm: math.div( $kendo-padding-y, 2) !default;
-$kendo-padding-x-lg: ( $kendo-padding-x * 1.5) !default;
-$kendo-padding-y-lg: ( $kendo-padding-y * 1.5) !default;
+$kendo-padding-sm-x: map.get( $kendo-spacing, 1 ) !default;
+$kendo-padding-sm-y: map.get( $kendo-spacing, 0.5 ) !default;
+$kendo-padding-md-x: map.get( $kendo-spacing, 2 ) !default;
+$kendo-padding-md-y: map.get( $kendo-spacing, 1 ) !default;
```
Comment (Juveniel) [inline packages/fluent/scss/core/_variables.scss:97]:
```suggestion
$kendo-padding-md-x: $kendo-padding-x !default;
$kendo-padding-md-y: $kendo-padding-y !default;
```

File: packages/fluent/scss/dataviz/_layout.scss [hunk 4]
```diff
@@ -99,8 +99,8 @@
     .k-chart-tooltip td {
         width: auto;
         text-align: left;
-        padding-inline: var( --kendo-padding-x-sm, #{$kendo-padding-x-sm} );
-        padding-block: var( --kendo-padding-y-sm, #{$kendo-padding-y-sm} );
+        padding-inline: var( --kendo-padding-x-sm, #{$kendo-padding-sm-x} );
+        padding-block: var( --kendo-padding-y-sm, #{$kendo-padding-sm-y} );
```
Comment (Juveniel) [inline packages/fluent/scss/dataviz/_layout.scss:103]: the css variable is `padding-x-sm`, while the scss var is `padding-sm-x`. We should probably unify the naming everywhere across components.

File: packages/fluent/scss/dataviz/_layout.scss [hunk 5]
```diff
@@ -302,8 +302,8 @@
 
         // Title
         .k-treemap-title {
-            padding-inline: var( --kendo-padding-x-sm, #{$kendo-padding-x-sm} );
-            padding-block: var( --kendo-padding-y-sm, #{$kendo-padding-y-sm} );
+            padding-inline: var( --kendo-padding-x-sm, #{$kendo-padding-sm-x} );
+            padding-block: var( --kendo-padding-y-sm, #{$kendo-padding-sm-y} );
```
Comment (Juveniel) [inline packages/fluent/scss/dataviz/_layout.scss:306]: same here for the naming discrepancy

File: packages/fluent/scss/dataviz/_layout.scss [hunk 6]
```diff
@@ -312,8 +312,8 @@
             background-repeat: repeat-x;
         }
         .k-treemap-title-vertical {
-            padding-inline: var( --kendo-padding-y-sm, #{$kendo-padding-y-sm} );
-            padding-block: var( --kendo-padding-x-sm, #{$kendo-padding-x-sm} );
+            padding-inline: var( --kendo-padding-y-sm, #{$kendo-padding-sm-y} );
+            padding-block: var( --kendo-padding-x-sm, #{$kendo-padding-sm-x} );
```
Comment (Juveniel) [inline packages/fluent/scss/dataviz/_layout.scss:316]: same here for the naming discrepancy

File: packages/fluent/scss/dataviz/_layout.scss [hunk 7]
```diff
@@ -344,16 +344,16 @@
...
         .k-treemap-title-vertical + .k-treemap-wrap {
-            inset-inline-start: calc( #{$kendo-padding-y-sm} * 2 + (#{$kendo-line-height} * #{$kendo-font-size}) + 1px);
+            inset-inline-start: calc( #{$kendo-padding-sm-y} * 2 + (#{$kendo-line-height} * #{$kendo-font-size}) + 1px);
         }
 
 
         // Leaf
         .k-leaf {
-            padding: var( --kendo-padding-x-sm, #{$kendo-padding-x-sm} );
+            padding: var( --kendo-padding-x-sm, #{$kendo-padding-sm-x} );
```
Comment (Juveniel) [inline packages/fluent/scss/dataviz/_layout.scss:356]: same here for the naming discrepancy

File: packages/fluent/scss/filter/_variables.scss [hunk 8]
```diff
@@ -2,10 +2,10 @@
 
 /// Horizontal padding of the filter.
 /// @group filter
-$kendo-filter-padding-x: $kendo-padding-x !default;
+$kendo-filter-padding-x: $kendo-padding-md-x !default;
 /// Vertical padding of the filter.
 /// @group filter
-$kendo-filter-padding-y: $kendo-padding-y !default;
+$kendo-filter-padding-y: $kendo-padding-md-y !default;
```
Comment (Juveniel) [inline packages/fluent/scss/filter/_variables.scss:8]: same here for the naming discrepancy

File: packages/fluent/scss/forms/_layout.scss [hunk 9]
```diff
@@ -226,24 +226,24 @@
...
+            padding-inline-end: var( --kendo-padding-x, #{$kendo-padding-md-x} );
             width: auto;
         }
 
         .k-form-field {
             display: flex;
             align-items: flex-start;
             text-align: left;
-            margin-block-end: calc( var( --kendo-padding-y-lg, #{$kendo-padding-y-lg} ) * 2 );
+            margin-block-end: calc( var( --kendo-padding-y-lg, #{$kendo-padding-lg-y} ) * 2 );
```
Comment (Juveniel) [inline packages/fluent/scss/forms/_layout.scss:239]: same here for the naming discrepancy

File: packages/fluent/scss/forms/_layout.scss [hunk 10]
```diff
@@ -226,24 +226,24 @@
...
 
             > span:not(.k-widget) {
                 width: var( --kendo-inline-form-element-width, #{$kendo-inline-form-element-width} );
                 text-align: right;
                 line-height: var( --kendo-line-height, #{$kendo-line-height} );
-                padding-block: calc( #{$kendo-padding-y} + #{$kendo-input-border-width} );
-                padding-inline-end: var( --kendo-padding-x-lg, #{$kendo-padding-x-lg} );
+                padding-block: calc( #{$kendo-padding-md-y} + #{$kendo-input-border-width} );
+                padding-inline-end: var( --kendo-padding-x-lg, #{$kendo-padding-lg-x} );
```
Comment (Juveniel) [inline packages/fluent/scss/forms/_layout.scss:246]: same here for the naming discrepancy

File: packages/material/scss/wizard/_variables.scss [hunk 11]
```diff
@@ -1,14 +1,13 @@
...
+$wizard-content-padding-y: $kendo-padding-md-y * 8 !default;
 $wizard-buttons-margin-x: 10px !default;
 
 $wizard-border-width: 0px !default;
 $wizard-font-size: $font-size !default;
 $wizard-line-height: $line-height !default;
 $wizard-font-family: $font-family !default;
 
-$wizard-step-border-focused: #656565 !default;
-$wizard-focused-shadow: inset 0 0 0 2px rgba( black, .13 ) !default;
```
Comment (Juveniel) [inline packages/material/scss/wizard/_variables.scss:14]: focus shadow removed here also

**Review by Juveniel — COMMENTED** [review]
For the size renaming in fluent especially, please verify that the generated css variable names in the _layout.scss foreach are using the same convention.

In terms of the breaking change note, apart from describing the prefixing with `$kendo`, we should also add note regarding the renamed variables ( state from `hovered` to `hover`, as well as other renamed variables ). Listing all of them should be ideal.

File: packages/default/scss/common/_base.scss (hunk 1 above)
Comment (joneff) [reply packages/default/scss/common/_base.scss:98]: ok. i will bring them back

File: packages/default/scss/wizard/_variables.scss (hunk 2 above)
Comment (joneff) [reply packages/default/scss/wizard/_variables.scss:14]: it was not used

File: packages/fluent/scss/dataviz/_layout.scss (hunk 4 above)
Comment (joneff) [reply packages/fluent/scss/dataviz/_layout.scss:103]: will do!

File: packages/material/scss/wizard/_variables.scss (hunk 11 above)
Comment (joneff) [reply packages/material/scss/wizard/_variables.scss:14]: it was not used

File: packages/fluent/scss/core/_variables.scss (hunk 3 above)
Comment (joneff) [reply packages/fluent/scss/core/_variables.scss:97]: The base padding variables are:

1) Unreliable -- they dont really make sense, as each theme is unique and cannot be properly transferred from one theme to another
2) Inconsistent -- sometimes, to achieve a certain fixed size, we multiply and divide and mix and match due to said uniqueness
3) Obsolete, as we are moving towards spacing maps

**Comment by joneff** [comment]
On a side note: the scheduler, inline editor and few other instances need their own variables. Ideally, a component, or a module, should only rely on it's own variables and rarely, if at all, should use global variables for styling.

**Review by Juveniel — APPROVED** [review]

**Comment by Juveniel** [comment]
@joneff you can squash the commits and add the BC info

**Pull request merged**